compiler/rustc_session/src/session.rs

```diff
@@ -1506,6 +1511,8 @@ pub fn build_session(
     let asm_arch =
         if target_cfg.allow_asm { InlineAsmArch::from_str(&target_cfg.arch).ok() } else { None };
 
+    let progress_bars = std::env::var_os("RUSTC_PROGRESS").map(|_| Session::init_progress_bars());
```

Comment:
Maybe check if the value is not `0` to allow disabling it using `RUSTC_PROGRESS=0`?